### PR TITLE
Rewrite ping server in Rust

### PR DESCRIPTION
**Pull request by swlynch99**
Add pingserver-rs which should be equivalent to pingserver but written in rust. In addition,
- Adds rust bindings for the various pelikan modules in rust-util
- Expand ccommon rust bindings
  - Only generate bindings for stuff that is actually a part of ccommon (instead of any part of libc that happens to be included)
  - Add some more high-level bindings to ccommon_rs

### CODE REVIEW

File: src/server/pingserver-rs/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+// This allows for a buffer to be allocated on the C side and freed on
+// the Rust side or vice-versa.
+#[global_allocator]
+pub static ALLOC: crate::alloc::LoggedAlloc = crate::alloc::LoggedAlloc;
+
+const LONG_USE_STR: &str = r#"\
+  pelikan_pingserver is, arguably, the most over-engineered 
+  ping server. 
+  
+  The prupose is to demonstrate how to create an otherwise
```
Comment (brayniac) [inline src/server/pingserver-rs/src/main.rs]: Typo

File: .gitignore [hunk 2]
```diff
@@ -46,5 +46,14 @@ cdb.rs
 compile_commands.json
 
 *.cmd
-core
+/core
 CMAKE_BINARY_DIR
+
+# Junk files from macos
+.DS_Store
```
[the remark this answers is not in the document]
Comment (swlynch99) [reply .gitignore]: I wasn't aware that you could do this. Fixed.

File: Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "src/storage/cdb/cdbgen",
+    "src/storage/cdb/cdb_rs",
+    "src/rust-util/pelikan"
+]
+
+[profile.release]
+opt-level = 3
+debug = true
+# rpath = false
+# lto = true
```
[the remark this answers is not in the document]
Comment (swlynch99) [reply Cargo.toml]: This was carried over from the Cargo.toml for ccommon. I've uncommented them now.

File: deps/ccommon/rust/ccommon_rs/src/buf.rs [hunk 4]
```diff
@@ -0,0 +1,115 @@
+
```
[the remark this answers is not in the document]
Comment (swlynch99) [reply deps/ccommon/rust/ccommon_rs/src/buf.rs]: Fixed

File: deps/ccommon/rust/ccommon_rs/src/buf.rs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+use cc_binding::buf;
+
+#[repr(transparent)]
+pub struct Buf {
```
[the remark this answers is not in the document]
Comment (swlynch99) [reply deps/ccommon/rust/ccommon_rs/src/buf.rs]: Fixed

**Comment by swlynch99** [comment]
Closing this so that it can be broken apart into multiple smaller prs

**Pull request closed without merging**